src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -237,7 +237,10 @@ public static MetadataReference MscorlibRef
             {
                 if (s_mscorlibRef == null)
                 {
-                    s_mscorlibRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_mscorlibRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll"),
+                        null);
```

Comment:
Is the same change needed for other `MetadataReference` properties in this file?

If so, perhaps we should have a helper: `GetMetadataReference(ref MetadataReference reference, string displayName, Func<MetadataReference> getMetadata)`.